src/Domain/LeanCode.DomainModels.EF/EFRepository.cs

```diff
@@ -58,4 +60,28 @@ public virtual void Update(TEntity entity)
     }
 
     public abstract Task<TEntity?> FindAsync(TIdentity id, CancellationToken cancellationToken = default);
+
+    [SuppressMessage(
+        "?",
+        "EF1001",
+        Justification = "There basically resembles the EntityFinder.FindTracked to mimic `FindAsync` snapshot re-use."
+    )]
+    protected TEntity? FindTracked(TIdentity id)
+    {
+        // Safety: aggregates are bound to have Id as a primary key by design.
+        var primaryKey = DbContext.Model.FindEntityType(typeof(TEntity))!.FindPrimaryKey()!;
+        return ((IDbContextDependencies)DbContext).StateManager!.TryGetEntryTyped(primaryKey, id)?.Entity as TEntity;
```

Comment:
`DbContext.GetService<IStateManager>()` could be more future-proof because it doesn't rely on the property in `IDbContextDependencies`. And it avoids an ugly cast.

Reply:
We trade readability (which would be bad no matter what way we take IMO) for potential "one lookup less" if the `StateManager` property is not implemented through the `GetRequiredService` call.